packages/dmn-editor/tests/e2e/drdArtifacts/deleteAssociationWaypoint.spec.ts

```diff
@@ -0,0 +1,135 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { expect } from "@playwright/test";
+import { test } from "../__fixtures__/base";
+import { DefaultNodeName, NodeType } from "../__fixtures__/nodes";
+import { TestAnnotations } from "@kie-tools/playwright-base/annotations";
+
+test.beforeEach(async ({ editor }) => {
+  await editor.open();
+});
+
+test.describe("Delete edge waypoint - Association", () => {
+  test.beforeEach(async ({ palette, nodes, browserName }) => {
+    test.skip(browserName === "webkit", "https://github.com/apache/incubator-kie-issues/issues/991");
+    test.info().annotations.push({
+      type: TestAnnotations.REGRESSION,
+      description: "https://github.com/apache/incubator-kie-issues/issues/991",
+    });
+
+    await palette.dragNewNode({ type: NodeType.INPUT_DATA, targetPosition: { x: 100, y: 100 } });
+    await nodes.dragNewConnectedNode({
+      from: DefaultNodeName.INPUT_DATA,
+      type: NodeType.TEXT_ANNOTATION,
+      targetPosition: { x: 100, y: 300 },
+    });
+  });
+
+  test("should delete the single Association edge waypoint", async ({ edges }) => {
+    await edges.addWaypoint({ from: DefaultNodeName.INPUT_DATA, to: DefaultNodeName.TEXT_ANNOTATION });
+
+    await edges.deleteWaypoint({
+      from: DefaultNodeName.INPUT_DATA,
+      to: DefaultNodeName.TEXT_ANNOTATION,
+      waypointIndex: 1,
+    });
+
+    await expect(
+      await edges.getWaypoint({
+        from: DefaultNodeName.INPUT_DATA,
+        to: DefaultNodeName.TEXT_ANNOTATION,
+        waypointIndex: 1,
+      })
+    ).not.toBeAttached();
+  });
+
+  test("should delete the single Association edge waypoint to make it straight", async ({ diagram, edges }) => {
+    await edges.addWaypoint({ from: DefaultNodeName.INPUT_DATA, to: DefaultNodeName.TEXT_ANNOTATION });
+    await edges.moveWaypoint({
+      from: DefaultNodeName.INPUT_DATA,
+      to: DefaultNodeName.TEXT_ANNOTATION,
+      waypointIndex: 1,
+      targetPosition: { x: 500, y: 300 },
+    });
+
+    await edges.deleteWaypoint({
+      from: DefaultNodeName.INPUT_DATA,
+      to: DefaultNodeName.TEXT_ANNOTATION,
+      waypointIndex: 1,
+    });
+    await expect(diagram.get()).toHaveScreenshot("delete-association-waypoint-straight-edge.png");
+  });
+
+  test("should delete all Association edge waypoints", async ({ nodes, edges }) => {
+    await edges.addWaypoint({ from: DefaultNodeName.INPUT_DATA, to: DefaultNodeName.TEXT_ANNOTATION });
+    await nodes.move({ name: DefaultNodeName.TEXT_ANNOTATION, targetPosition: { x: 200, y: 500 } });
+
+    await edges.addWaypoint({ from: DefaultNodeName.INPUT_DATA, to: DefaultNodeName.TEXT_ANNOTATION });
```

Comment:
As the "GIVEN" part is very simple, I guess we could group them instead of breaking a new line.

Reply:
yes, sounds good, will do for all "should delete all ..."